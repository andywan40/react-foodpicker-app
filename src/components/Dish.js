import React, { useEffect } from "react";
import { Link } from "react-router-dom";
import { ToastContainer, toast } from "react-toastify";
import { withStyles } from "@material-ui/styles";
import styles from "../styles/DishStyles";

function Dish({ classes, history, match, location }) {
  const { cuisine, dish } = match.params;
  const dishInfo =
    location?.dishProps?.dishInfo ||
    JSON.parse(sessionStorage.getItem(`${dish.toLowerCase()}-dishInfo`));
  if (!dishInfo) {
    toast.error(`Can't Fetch ${dish} Data`, {
      position: "top-right",
      autoclose: 2500,
      hideProgressBar: false,
      closeOnClick: true,
      pauseOnHover: true,
      draggable: true,
      progress: undefined,
      toastId: "custom",
      onClose: () => history.push("/"),
    });
  }

  useEffect(() => {
    sessionStorage.setItem(
      `${dish.toLowerCase()}-dishInfo`,
      JSON.stringify(dishInfo)
    );
  }, [dishInfo, dish]);

  const handleSearch = () => {
    window.open(`https://www.google.com/search?q=${dishInfo.title}+near+me`);
  };

  const buttonDivStyles = !dishInfo?.canGetRecipe
    ? {
        gridTemplateColumns: "repeat(2, 60%)",
      }
    : {};

  return (
    <div className={classes.root} data-testid="dish">
      <ToastContainer />
      {dishInfo && (
        <div className={classes.contentContainer}>
          <h2
            className={classes.title}
            onClick={handleSearch}
            data-testid="title"
          >
            {dishInfo.title}
          </h2>
          <h4 className={classes.desc}>{dishInfo.desc}</h4>
          <img
            className={classes.img}
            src={dishInfo.url}
            alt={dishInfo.alt}
            data-testid="img"
          />
          <h6>
            Photo by{" "}
            <a
              href={`https://unsplash.com/@${dishInfo.username}?utm_source=foodpickerapp&utm_medium=referral`}
            >
              {dishInfo.user}
            </a>{" "}
            on
            <a href="https://unsplash.com/?utm_source=foodpickerapp&utm_medium=referral">
              {" "}
              Unsplash
            </a>
          </h6>
          <div className={classes.buttonDiv} style={buttonDivStyles}>
            <Link to={`/${cuisine}`} className={classes.button1}>
              {cuisine} food
            </Link>
            {dishInfo.canGetRecipe && (
              <Link
                to={`/recipelist/${cuisine}/${dishInfo.title}`}
                className={classes.button2}
              >
                Get Recipes
              </Link>
            )}
            <Link to="/" className={classes.button2}>
              home
            </Link>
          </div>
        </div>
      )}
    </div>
  );
}

export default withStyles(styles)(Dish);
