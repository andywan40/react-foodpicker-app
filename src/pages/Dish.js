import React from "react";
import { Link } from "react-router-dom";
import { withStyles } from "@material-ui/styles";
import styles from "../styles/DishStyles";

function Dish(props) {
  const { classes } = props;
  const cuisine = props.match.params.cuisine;
  const dish = props.match.params.dish;
  const dishInfo =
    props.location?.dishProps?.dishInfo ||
    JSON.parse(sessionStorage.getItem("dishInfo"));
  sessionStorage.setItem("dishInfo", JSON.stringify(dishInfo));
  const handleSearch = () => {
    window.open(`https://www.google.com/search?q=${dish}+near+me`);
  };
  const goBackLink = `/${cuisine}`;
  const usernameLink = `https://unsplash.com/@${dishInfo.username}?utm_source=foodpickerapp&utm_medium=referral`;
  const unsplashLink =
    "https://unsplash.com/?utm_source=foodpickerapp&utm_medium=referral";
  return (
    <div className={classes.root}>
      <div className={classes.contentContainer}>
        <h2 className={classes.title} onClick={handleSearch}>
          {dish}
        </h2>
        <h4>{dishInfo.desc}</h4>
        <img className={classes.img} src={dishInfo.url} alt={dishInfo.alt} />
        <h6>
          Photo by <a href={usernameLink}>{dishInfo.user}</a> on
          <a href={unsplashLink}> Unsplash</a>
        </h6>
        <div className={classes.buttonDiv}>
          <Link to={goBackLink}>
            <button className={classes.button1}>{cuisine} food</button>
          </Link>
          <Link to="/">
            <button className={classes.button2}>back</button>
          </Link>
        </div>
      </div>
    </div>
  );
}

export default withStyles(styles)(Dish);
