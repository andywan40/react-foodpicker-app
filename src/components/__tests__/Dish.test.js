import {render, screen, cleanup} from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Dish from '../Dish';

afterEach(()=> {
    cleanup();
})

test("should render component", () => {
    render(
        <MemoryRouter initialEntries={["/american/grits"]}>
            <Dish match={{params: {cuisine: "american", dish: "grits"}}} location={{dishProps: { dishInfo: { canGetRecipe: true, title: "gritsss", url: "testurl"}}}}/>
        </MemoryRouter>
    );
    // const dish = screen.getByTestId("dish");
    // expect(dish).toBeInTheDocument();
    const title = screen.getByTestId("title");
    expect(title.textContent).toBe("gritsss");
    //const img = screen.getByTestId("img");
    //expect(img.getAttribute("src")).toBe("testurl");
})


test("matches snapshot", ()=> {
    const tree = renderer.create(
        <MemoryRouter initialEntries={["/american/grits"]}>
            <Dish match={{params: {cuisine: "american", dish: "grits"}}} location={{dishProps: { dishInfo: { canGetRecipe: true, title: "gritsss", url: "testurl"}}}}/>
        </MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})