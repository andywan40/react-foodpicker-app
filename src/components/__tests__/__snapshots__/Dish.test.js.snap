// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Dish-root-9"
  data-testid="dish"
>
  <div
    className="Toastify"
  />
  <div
    className="Dish-contentContainer-10"
  >
    <h2
      className="Dish-title-11"
      data-testid="title"
      onClick={[Function]}
    >
      gritsss
    </h2>
    <h4 />
    <a
      href="/recipelist/american/gritsss"
      onClick={[Function]}
    >
      <button
        className="Dish-recipeButton-12"
      >
        Get Recipes
      </button>
    </a>
    <img
      className="Dish-img-13"
      data-testid="img"
      src="testurl"
    />
    <h6>
      Photo by 
      <a
        href="https://unsplash.com/@undefined?utm_source=foodpickerapp&utm_medium=referral"
      />
       on
      <a
        href="https://unsplash.com/?utm_source=foodpickerapp&utm_medium=referral"
      >
         Unsplash
      </a>
    </h6>
    <div
      className="Dish-buttonDiv-14"
    >
      <a
        href="/american"
        onClick={[Function]}
      >
        <button
          className="Dish-button1-15"
        >
          american
           food
        </button>
      </a>
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="Dish-button2-16"
        >
          home
        </button>
      </a>
    </div>
  </div>
</div>
`;
